csp/middleware.py

```diff
@@ -34,11 +35,15 @@ class CSPMiddleware(MiddlewareMixin):
     See http://www.w3.org/TR/CSP/
 
     """
-    def _make_nonce(self, request, length=16):
+    def _make_nonce(self, request):
         # Ensure that any subsequent calls to request.csp_nonce return the
         # same value
         if not getattr(request, '_csp_nonce', None):
-            request._csp_nonce = get_random_string(length)
+            request._csp_nonce = (
+                base64
+                .b64encode(os.urandom(16))
+                .decode("ascii")
+            )
```

Comment:
if we can drop support for python <3.6 or django <3.0 we can use:

```suggestion
            request._csp_nonce = secrets.token_urlsafe(16)
```

Reply:
because csp nonce recommends base64url or base64

Reply:
For the secrets method html5 validator says Content-Security-Policy HTTP header: Bad content security policy: Invalid base64-value (should be multiple of 4 bytes: 22). 

Probably because it trims the ==

Reply:
Yup base64ing per the original is best